src/api-service/__app__/onefuzzlib/workers/nodes.py

```diff
@@ -267,15 +267,15 @@ def can_process_new_work(self) -> bool:
                 "can_schedule is set to be deleted.  machine_id:%s",
                 self.machine_id,
             )
-            self.stop()
+            self.stop(done=True)
```

Comment:
Could we be introducing a racing condition here ? 
The node could have picked up a new item from the work queue before receiving the done command.
In that case we could be loosing some work.
same for the `reimage_requested` case below

Reply:
I don't think so.

`can_process_new_work` is called from the supervisor's endpoint verifying that work from the queue can be run.  In the places we're updating (checking `node.is_outdated`, `node.delete_requested`, and `node.reimage_requested`) are node specific responses.  All worksets checked for this node would fail until the node is reimaged.